doc/guides/arch/arm_cortex_m.rst

```diff
@@ -0,0 +1,643 @@
+   * the PSP of the current thread which points to beginning of the current thread's exception
+     stack frame. The latter contains the caller-saved context and the return address of the switched-out
+     thread.
+
+* When switching-in a new thread the processor
+
+   * restores the new thread's callee-saved registers from the thread's
+     container for callee-saved registers
+   * restores the new thread's operation *mode*
+   * restores the FP callee-saved registers if the switched-in thread have had
+     an active FP context before being switched-out
+   * re-programs the dynamic MPU regions to allow a user thread access its stack and application
+     memories, and/or programs a stack-overflow MPU guard at the bottom of the thread's
+     privileged stack.
+   * restores the PSP for the incoming thread and re-programs the stack pointer limit
+     register (if applicable, see :kconfig:`CONFIG_BUILTIN_STACK_GUARD`)
+   * optionally does a stack limit checking for the switched-in thread, if
+     sentinel-based stack limit checking is enabled (see :kconfig:`CONFIG_STACK_SENTINEL`)
+
+PendSV exception return sequence restores the new thread's caller-saved registers and the
+return address, as part of unstacking the exception stack frame.
+
+The implementation of the context-switch mechanism is present in
+:file:`arch/arm/core/aarch32/swap_helper.S`.
+
+Stack limit checking (Armv8-M)
+------------------------------
+
+Armv8-M and Armv8.1-M variants support stack limit checking using the MSPLIM and PSPLIM
+core registers. The feature is enabled when :kconfig:`CONFIG_BUILTIN_STACK_GUARD` is set.
+When stack limit checking is enabled, both the thread's privileged or user stack, as well
+as the interrupt stack are guarded by PSPLIM and MSPLIM registers, respectively. MSPLIM is
+configured *once* during kernel boot, while PSLIM is re-programmed during every thread
+context-switch or during system calls, when the thread switches from using its default
+stack to using its privileged stack, and vice versa. PSPLIM re-programming
+
+* has a relatively low runtime overhead (programming is done with MSR instructions)
+* does not impact interrupt latency
+* does not require any memory areas to be reserved for stack guards
+* does not make use of MPU regions
+
+It is, therefore, considered as a lightweight but very efficient stack overflow
+detection mechanism in Cortex-M applications.
+
+Stack overflows trigger the dedicated UsageFault exception provided by Armv8-M.
+
+Interrupt handling features
+===========================
+
+This section describes certain aspects around exception and interrupt
+handling in Arm Cortex-M.
+
+Interrupt priority levels
+-------------------------
+
+The number of available (configurable) interrupt priority levels is
+determined by the number of implemented interrupt priority bits in
+NVIC; this needs to be described for each Cortex-M platform using
+DeviceTree:
+
+   .. code-block:: c
```

Comment:
```suggestion
.. code-block:: c
```

Reply:
Yeap, just realized this is not c code :) 
This was also leading to a build failure. I have changed to console, but perhaps devicetree is a better match

Reply:
done now